sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/SessionReceiverManager.cs

```diff
@@ -410,13 +406,18 @@ protected override async Task RaiseExceptionReceived(ProcessErrorEventArgs event
             }
         }
 
-        internal void CancelSession()
+        internal async Task CancelSessionAsync()
         {
             if (_sessionCancellationSource is { IsCancellationRequested: false })
             {
                 _sessionCancellationSource.Cancel();
                 _sessionCancellationSource.Dispose();
             }
+
+            if (_sessionLockRenewalTask != null)
```

Comment:
Alternatively, we could consider an exception block here that logs to avoid unobserved exceptions when we're fire-n-forget calling.

Reply:
The lock renewal task itself has logging for non TCE - https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/SessionReceiverManager.cs#L378

For TCE, these are expected and therefore not logged.